test/test_type_promotion.py

```diff
@@ -218,6 +218,20 @@ def test_create_bool_tensors(self):
         # this seems like odd behavior but ints also create float tensors, numpy doesn't have this function.
         self.assertEqual(torch.scalar_tensor(False, device=self.device), torch.tensor(0., device=self.device))
 
+    def test_comparison_ops_with_type_promotion(self):
```

Comment:
These set of tests are not sufficient to comprehensively test the type promotion behavior. I'd like to see the following test cases (some of these are probably redundant, I'm just listing off what comes to mind)
- torch.lt(int8, int32)   (left operand gets promoted)
- torch.lt(int32, int8)   (right operand gets promoted)
- left operand gets promoted, out= left
- left operand gets promoted, out= right
- right operand gets promoted, out= right
- right operand gets promoted, out= left
- int tensor < float
- int tensor < float , out= int tensor
- float < int tensor
- float < int tensor, out = float tensor.

In addition, it looks like there's a lot of infrastructure in this file for performing testing already. @nairbv is there a way to adapt the existing tests for our soon-to-exist comparison ops?

Reply:
All the possible test cases were tested. We do not allowed to out in something other then Bool or Byte tensor

Reply:
I think you're still missing the following cases:
- torch.lt(int8, int32) (left operand gets promoted) (out-of-place)
- torch.lt(int32, int8) (right operand gets promoted) (out-of-place)
- float < int tensor